File: src/main/kotlin/starry/memebrowser/entity/Meme.kt
package starry.memebrowser.entity

import kotlinx.serialization.Serializable

@Serializable
data class Meme(
    val id: String,
    val title: String,
    val imageUrl: String,
    val originalText: String,
    val recognizedText: String? = null,
    val content: String,
    val tags: Set<String>,
    val createdAt: String
) {
    fun tags(vararg newTags: String) = copy(tags = tags + newTags)
}

File: src-vue/src/types/meme.ts
export interface Meme {
  id: string;
  title: string;
  imageUrl: string;
  originalText: string;
  recognizedText?: string; // 图片中识别到的文本
  content: string;
  tags: string[];
  createdAt: string;
}

export interface MemeResponse {
  data: Meme[];
  total: number;
  page: number;
  limit: number;
  hasMore: boolean;
}

export interface SearchParams {
  query?: string;
  tags?: string[];
  tag?: string;
  page?: number;
  limit?: number;
}
